ase/Makefile

```diff
@@ -517,8 +517,18 @@ ifdef DUT_VLOG_SRC_LIST
 	cd $(WORK) ; vlog $(MENT_VLOG_OPT) $(ASE_PLATFORM_INC) -F $(DUT_VLOG_SRC_LIST) -l vlog-afu.log
 endif
 
+$(WORK):
+	mkdir -p $(WORK)
+	@# Link to HEX memory images generated and required by Qsys
+	@if [ -f qsys_hex_sim_files.list ]; then \
+	  for h in `cat qsys_hex_sim_files.list`; do \
+	    echo Linking to $${h} ; \
+	    ln -s ../$${h} work ; \
```

Comment:
Is it possible that the "work" lib could be linked by multiple hex memory image files here?

Reply:
I assume you mean multiple hex files sharing the same name. No. The sim_files_found set guarantees that all file names are unique. This is a Qsys property -- that files with the same name anywhere in the tree have the same contents. Qsys adds hashes to generated file names when necessary to preserve this constraint.

Reply:
Isn't the "work" pointing to different hex image files here? Not sure how Qsys works.

Reply:
No. Links are created in the ASE work directory to hex files in the Qsys-generated tree. You'd think that Qsys would generate names with absolute paths, or do something clever to load hex files relative to the source files that open them. It does neither, instead opening .hex files in the work directory.

Note that the local memory emulation already does the same thing in the Makefile.

Reply:
One more generic questions regarding the Quartus sim library: all AFU will linked to all the simulation lib in Quartus no matter it needs it or not?

Reply:
Correct. This has not changed. There is no good way to predict when it will be needed.

Reply:
Thanks.